File: src/i18n/locales/sv.ts
import type { Locale } from "../i18n.js";

export const sv: Locale = {
  validation: {
    weight: {
      failedToParse: "Misslyckades med att tolka viktformatet",
      invalidWeightNumberFormat:
        "Kunde inte känna igen viktformatet. Förväntade ett två- eller tresiffrigt nummer, eventuellt med decimaldel.",
      weightTooLow: "Vikten kan inte vara mindre än 10 kg.",
      weightTooHigh: "Vikten kan inte vara mer än 200 kg.",
    },
    timestamp: {
      notString: "Tidsstämpeln är inte en sträng.",
      notDate: "Kunde inte känna igen datumformatet.",
      failedToParseStart: "Misslyckades med att tolka startdatumformatet",
      failedToParseEnd: "Misslyckades med att tolka slutdatumformatet",
    },
    auth: {
      failedToParse: "Misslyckades med att tolka användardataformatet",
      invalidFormat: "Användarnamn eller lösenord är inte en sträng",
    },
  },
  response: {
    weight: {
      addSuccess: "Vikt tillagd framgångsrikt",
    },
    migration: {
      success: "Migrering slutförd framgångsrikt",
    },
    user: {
      registerSuccess: "Användare registrerad framgångsrikt",
    },
  },
  error: {
    connection: {
      notSet: "Databasanslutning inte etablerad",
    },
    user: {
      exists: "Användare med detta namn finns redan",
      hashFailed: "Misslyckades med att hasha lösenordet",
      failedToAuthorize: "Auktorisering misslyckades",
      unauthorized: "Användaren är inte auktoriserad",
    },
    unknown: "Ett okänt fel uppstod",
  },
};

File: go/internal/i18n/locales/sv.go
package locales

var Sv = Locale{
	Validation: ValidationLocale{
		Weight: WeightValidationLocale{
			FailedToParse:             "Misslyckades med att tolka viktformatet",
			InvalidWeightNumberFormat: "Kunde inte känna igen viktformatet. Förväntades ett två- eller tresiffrigt tal, eventuellt med en decimaldel.",
			WeightTooLow:              "Vikten kan inte vara mindre än 10 kg.",
			WeightTooHigh:             "Vikten kan inte vara mer än 200 kg.",
		},
		Timestamp: TimestampValidationLocale{
			NotString:          "Tidsstämpeln är inte en sträng.",
			NotDate:            "Kunde inte känna igen datumformatet.",
			FailedToParseStart: "Misslyckades med att tolka startdatumformatet",
			FailedToParseEnd:   "Misslyckades med att tolka slutdatumformatet",
		},
		Auth: AuthValidationLocale{
			FailedToParse: "Misslyckades med att tolka användardataformatet",
			InvalidFormat: "Användarnamn eller lösenord är inte en sträng",
		},
	},
	Response: ResponseLocale{
		Weight:    "Vikt tillagd framgångsrikt",
		Migration: "Migrering slutförd framgångsrikt",
		User:      "Användare registrerad framgångsrikt",
	},
	Error: ErrorLocale{
		Connection: ConnectionErrorLocale{
			NotSet: "Databasanslutning inte etablerad",
		},
		User: UserErrorLocale{
			Exists:            "Användare med detta namn finns redan",
			HashFailed:        "Misslyckades med att hasha lösenordet",
			FailedToAuthorize: "Auktorisering misslyckades",
			Unauthorized:      "Användaren är inte auktoriserad",
		},
		Unknown: "Ett okänt fel uppstod",
	},
}
